config/routes.rb

```diff
@@ -77,7 +77,7 @@
         get '/login/two_factor/piv_cac' => 'two_factor_authentication/piv_cac_verification#show'
       end
       get  '/login/two_factor/:otp_delivery_preference' => 'two_factor_authentication/otp_verification#show',
-           as: :login_two_factor
+           as: :login_two_factor, constraints: { otp_delivery_preference: /sms|voice/ }
```

Comment:
should we make the regex beginning of string, end of string to be safe?

Reply:
I know it's not needed and verified that xsmsx doesn't work. But I remember banging my head trying to understand why years ago.  Is the beginning of string end of string implied when it uses constraints?

Reply:
It's obviously not doing regular regex